exchanges/gateio/gateio.go

```diff
@@ -424,169 +1013,2938 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
+	var response []TradingHistoryItem
+	params := url.Values{}
+	settle = strings.ToLower(settle)
+	if settle != settleBTC && settle != settleUSD && settle != settleUSDT {
+		return nil, errMissingSettleCurrency
+	}
+	if contract.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
+	}
+	contract.Delimiter = currency.UnderscoreDelimiter
+	params.Set("contract", contract.Upper().String())
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if offset > 0 {
+		params.Set("offset", strconv.FormatUint(offset, 10))
+	}
+	if lastID != "" {
+		params.Set("last_id", lastID)
+	}
+	if !from.IsZero() {
+		params.Set("from", strconv.FormatInt(from.Unix(), 10))
+	}
+	if !to.IsZero() {
+		params.Set("to", strconv.FormatInt(to.Unix(), 10))
+	}
+	return response, g.SendHTTPRequest(ctx, exchange.RestSpot,
+		common.EncodeURLValues(fmt.Sprintf(futuresTrades, settle), params), &response)
+}
+
+// GetFuturesCandlesticks retrives specified contract candlesticks.
+func (g *Gateio) GetFuturesCandlesticks(ctx context.Context, settle string, contract currency.Pair, from, to time.Time, limit uint64, interval kline.Interval) ([]FuturesCandlestick, error) {
+	var candlesticks []FuturesCandlestick
+	params := url.Values{}
+	settle = strings.ToLower(settle)
+	if settle != settleBTC && settle != settleUSD && settle != settleUSDT {
+		return nil, errMissingSettleCurrency
+	}
+	if contract.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
+	}
+	fPair, err := g.GetPairFormat(asset.Futures, true)
+	if err != nil {
+		fPair = currency.PairFormat{
+			Delimiter: currency.UnderscoreDelimiter,
+			Uppercase: true,
+		}
+	}
+	pairString := strings.ToUpper(fPair.Format(contract))
+	params.Set("contract", pairString)
+	if !from.IsZero() {
+		params.Set("from", strconv.FormatInt(from.Unix(), 10))
+	}
+	if !to.IsZero() {
+		params.Set("to", strconv.FormatInt(to.Unix(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if intervalString := g.GetIntervalString(interval); intervalString != "" {
+		params.Set("interval", intervalString)
+	}
+	return candlesticks, g.SendHTTPRequest(ctx, exchange.RestSpot,
+		common.EncodeURLValues(fmt.Sprintf(futuresCandlesticks, settle), params),
+		&candlesticks)
+}
+
+// GetFuturesTickers retrives futures ticker information for a specific settle and contract info.
+func (g *Gateio) GetFuturesTickers(ctx context.Context, settle string, contract currency.Pair) ([]FuturesTicker, error) {
+	var tickers []FuturesTicker
+	params := url.Values{}
+	settle = strings.ToLower(settle)
+	if settle != settleBTC && settle != settleUSD && settle != settleUSDT {
+		return nil, errMissingSettleCurrency
+	}
+	if !contract.IsEmpty() {
+		fPair, err := g.GetPairFormat(asset.Futures, true)
+		if err != nil {
+			fPair = currency.PairFormat{
+				Delimiter: currency.UnderscoreDelimiter,
+				Uppercase: true,
+			}
+		}
```

Comment:
There is a lot of duplication code here across this file, `I suggest` we could have a separate method, but technically we should already be calling this method with a correctly formatted currency.Pair for that asset.